File: 2144_Minimum_Cost_of_Buying_Candies_With_Discount/solution.c
int comp(const void* a, const void* b) { return (*(int*)b - *(int*)a); }
int minimumCost(int* cost, int n) {
    qsort(cost, n, sizeof(int), comp);
    int r = 0;
    for (int i = 0; i < n; ++i) if (i % 3 != 2) r += cost[i];
    return r;
}

File: 2144_Minimum_Cost_of_Buying_Candies_With_Discount/solution.py
class Solution:
    def minimumCost(self, cost: List[int]) -> int:
        cost.sort(key=lambda x: -x)
        r = 0
        n = len(cost)
        for i in range(n):
            if i % 3 != 2:
                r += cost[i]
        return r
